dm/dm/ctl/master/check_task.go

```diff
@@ -63,10 +65,26 @@ func checkTaskFunc(cmd *cobra.Command, _ []string) error {
 		return err
 	}
 
+	// we check if `is-sharding` is explicitly set, to distinguish between `false` from default value
+	isShardingSet := false
+	lines := bytes.Split(content, []byte("\n"))
+	for i := range lines {
+		if bytes.HasPrefix(lines[i], []byte("is-sharding")) {
+			isShardingSet = true
+			break
+		}
+	}
+	task := config.NewTaskConfig()
+	yamlErr := task.RawDecode(string(content))
+	// if can't parse yaml, we ignore this check
+	if yamlErr == nil && isShardingSet && !task.IsSharding && task.ShardMode != "" {
```

Comment:
what if `yamlErr != nil` , we can ignore this?

Reply:
keep old behaviour